android/BOINC/app/build.gradle

```diff
@@ -13,38 +12,53 @@ import org.ajoberstar.grgit.Grgit
 def buildVersionCode() {
     def repo = Grgit.open(currentDir: project.rootDir)
     def head = repo.head()
-
-    // Sanity check across git plugins
-    assert head.getAbbreviatedId(10) == versionDetails().gitHash: "Internal error: SHA1 mismatch!"
-
-    return head.time
+    def versionCode = head.time
+    println "versionCode: " + versionCode
+    return versionCode
 }
 
 // Derive version name from release tag and add commit SHA1
 def buildVersionName() {
-    def pattern = /client_release\/\d+\.\d+\/(?<major>\d+)\.(?<minor>\d+)\.(?<revision>\d+)(?<suffix>[-_\.]?.*)/
+    def pattern = /(?<tag>.+)-(?<offset>\d+)-g(?<hash>[0-9a-f]+)/
     def version = ': DEVELOPMENT'
+    def isDev = true
+    def offset = -1
+    def tag = ""
 
-    def head = versionDetails()
-    def tag = head.lastTag
-    def match = (tag =~ pattern)
-
-    // Sanity checks for tag format
-    if (match.hasGroup() && 1L == match.size() && 5L == match[0].size() && head.commitDistance == 0) {
-        def major = match.group('major')
-        def minor = match.group('minor')
-        def revision = match.group('revision')
-        def suffix = match.group('suffix')
-        version = "${major}.${minor}.${revision}${suffix}"
-        assert !suffix.endsWith('.dirty'): "Dirty working tree detected! Preventing release build!"
-    } else {
-        println "Warning! Non-release tag or offset found: $tag (offset: $head.commitDistance)"
-        println "Flagging as DEVELOPMENT build..."
-    }
+    def repo = Grgit.open(currentDir: project.rootDir)
+    def head = repo.head()
+    def tagWithOffset = repo.describe(longDescr:true ,tags: true)
```

Comment:
remove space before comma and add space after comma

Reply:
Fix